persistence/src/main/java/de/cyface/persistence/DatabaseHelper.java

```diff
@@ -89,44 +90,51 @@ public void onCreate(final SQLiteDatabase db) {
 
     /**
      * The onUpgrade method is called when the app is upgraded and the DATABASE_VERSION changed.
-     * The incremental database upgrades are executed to reach the current version.
+     * <p>
+     * This method is not called incrementally by the system which is why this method implements this.
+     * <p>
+     * This method is automatically executed in a transaction, do not wrap the code in another transaction!
      *
      * @param database the database which shall be upgraded
      * @param oldVersion the database version the app was in before the upgrade
      * @param newVersion the database version of the new, upgraded app which shall be reached
      */
     @Override
     public void onUpgrade(final SQLiteDatabase database, final int oldVersion, final int newVersion) {
-        Log.w(TAG, "Upgrading database from version " + oldVersion + " to " + newVersion);
+        Validate.isTrue(oldVersion == 8 || oldVersion >= 11, "Unsupported versions");
```

Comment:
The versions 9 and 10 were never released to outsiders.
We also don't support upgrades from < 8 as the first 2019 STAD release uses V8.